docker/api/build.py

```diff
@@ -11,7 +12,47 @@
 
 log = logging.getLogger(__name__)
 
-
+class FileUploadBuffer(object):
+    '''
+    Wrap a file object, buffer, and print progress.
+    '''
+    BUFFER_SIZE = 1024*32
+    def __init__(self, file):
+        self.read_so_far = 0
+        self.num_reads = 0
+        self.starttime = datetime.datetime.utcnow()
+        self.file = file
+    
+    def __iter__(self, *args, **kwargs):
+        self.starttime = datetime.datetime.utcnow()
+        return self
+    
+    def chk_print(self, fn):
+        '''
+        Debug function that prints out how long we've been uploading.
+        '''
+        old_megs = self.read_so_far / 1024 / 1024
+        out = fn()
+        self.num_reads += 1
+        self.read_so_far += len(out)
+        new_megs = self.read_so_far / 1024 / 1024
+        if old_megs != new_megs:
+            log.debug('Sent %sMB, %d file reads, time elapsed: %s'%(new_megs,
+                self.num_reads, datetime.datetime.utcnow() - self.starttime))
+        return out
+        
+    def next(self):
+        out = self.read(self.BUFFER_SIZE)
+        if not len(out):
+            raise StopIteration
+        return out
+        
+    def read(self, *args, **kwargs):
+        return self.chk_print(fn = lambda: self.file.read(*args, **kwargs))
```

Comment:
This function seems unnecessary. Why not just move the implementation of `chk_print` here?